src/commands/kv/write_bulk.rs

```diff
@@ -0,0 +1,111 @@
+extern crate base64;
+
+use cloudflare::apiclient::ApiClient;
+use percent_encoding::{percent_encode, PATH_SEGMENT_ENCODE_SET};
+use walkdir::WalkDir;
+
+use std::ffi::OsString;
+use std::fs;
+use std::fs::metadata;
+use std::path::Path;
+
+use failure::bail;
+use cloudflare::workerskv::write_bulk::BulkWrite;
+use cloudflare::workerskv::write_bulk::BulkWriteParams;
+use cloudflare::workerskv::write_bulk::KeyValuePair;
+
+use crate::terminal::message;
+
+pub fn write_bulk(namespace_id: &str, filename: &Path, expiration: Option<&str>, ttl: Option<&str>) -> Result<(), failure::Error> {
+    let client = super::api_client()?;
+    let account_id = super::account_id()?;
+
+    // If the provided argument for write_bulk is a json file, parse it 
+    // and upload its contents. If the argument is a directory, create key-value
+    // pairs where keys are the relative pathnames of files in the directory, and
+    // values are the base64-encoded contents of those files. 
+    let mut data;
+    let pairs: Result<Vec<KeyValuePair>, failure::Error> = match metadata(filename) {
+        Ok(ref file_type) if file_type.is_file() => {
+            data = fs::read_to_string(filename)?;
+            Ok(serde_json::from_str(&data)?)
+        }
+        Ok(ref file_type) if file_type.is_dir() => {
+            parse_directory(filename)
+        }
+        Ok(_file_type) => { // any other file types (namely, symlinks)
+            bail!("Cannot upload a file that is a symlink: {}", filename.display())
+        }
+        Err(e) => bail!(e)
+    };
+
+    let response = client.request(&BulkWrite {
+        account_identifier: &account_id,
+        namespace_identifier: namespace_id,
+        bulk_key_value_pairs: pairs?,
+
+        params: BulkWriteParams {
+            expiration: expiration,
+            expiration_ttl: ttl,
+        },
+    });
+
+    match response {
+        Ok(_success) => message::success(&format!("Success")),
+        Err(e) => super::print_error(e),
+    }
+
+    Ok(())
+}
+
+fn parse_directory(directory: &Path) -> Result<Vec<KeyValuePair>, failure::Error> {
```

Comment:
I foresee there will be common errors with uploading values that are over the `2MB` limit. How do we want to handle this? (actually, what does our API do right now? does the whole write fail?)

Reply:
This functionality is being added in an upcoming commit, keep your eyes out!

Reply:
We're going to fall back on API responses for this, at least for the time being.